Engineering/UnitTesting.md

```diff
@@ -1,41 +1,56 @@
 # Unit Testing
 
 ## Goals
-Unit tests play an integral role in building quality software and enabling agile methodologies. [CSE](../CSE.md) recommends all efforts follow [Test Driven Development](http://deviq.com/test-driven-development/) where ever possible, i.e. code should have unit tests unless it's developed for an environment with unit testing capabilities, e.g. Azure Stream Analytics. With TDD, engineers start with coding the test(s), which will initially fail. The implementation of the unit is finished when the unit satisfies the tests.
+Unit tests play an integral role in building quality software and enabling agile methodologies. [CSE](../CSE.md) recommends all efforts follow [Test Driven Development](http://deviq.com/test-driven-development/) where ever possible, i.e. code should have unit tests unless it's developed for an environment with unit testing capabilities, e.g. Azure Stream Analytics. 
+
+With TDD, an engineer follows the pattern of RED/GREEN/REFACTOR to build only the code required and nothing more.
+- RED: Start with a test to exercise a function. Initially, this test will fail because the function is not yet implemented (or is incomplete).
+- GREEN: Implements just enough code to get all of the currently written tests to pass.
+- REFACTOR: Modify the code as necessary to reflect for naming standards, remove code duplication and implement clean coding practices.
+
+Development continues in this cycle (RED/GREEN/REFACTOR) writing additional tests until all functional requirements are complete.
 
 ### Unit tests have several goals:
-- Ensure code fulfills functional and non-functional requirements
-- Ensure focus on functionality to deliver
-- Support fast code evolution and refactoring while reducing the risk of regressions
-- Provide confidence to potential contributors 
+- Ensure code fulfills functional requirements. 
+- Ensure focus is delivering required functionality not unnecessary requirements.
+- Support fast code evolution and refactoring while reducing the risk of regressions.
+- Provide confidence to potential contributors.
 - Developer Documentation of API usage
 
+>Note: Non-functional requiements should be validated with integration tests (i.e. Server response time in less than 2 seconds.)
+
```

Comment:
Same as Beat's comment below. If the scope is the unit, then those would be part of unit tests